compat_test/smoke_compat_test.go

```diff
@@ -0,0 +1,193 @@
+
+	initCtx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	t.Cleanup(cancel)
+
+	// Fetch Chain IDs and make sure they're same
+	celoChainId, err := celoEthClient.ChainID(initCtx)
+	require.NoError(t, err)
+	opChainId, err := opEthClient.ChainID(initCtx)
+	require.NoError(t, err)
+	require.Equal(t, celoChainId.Uint64(), opChainId.Uint64(), "chain ids of referenced chains differ")
+
+	// Make sure the Chain ID is supported
+	chainId := celoChainId.Uint64()
+	_, ok := gingerbreadBlocks[chainId]
+	require.True(t, ok, "chain id %d not found in supported chainIDs %v", chainId, gingerbreadBlocks)
+
+	// Fetch the last block height in Celo L1
+	lastCeloL1BlockHeight, err := celoEthClient.BlockNumber(initCtx)
+	require.NoError(t, err)
+
+	t.Logf("Last Celo L1 Block Height: %d", lastCeloL1BlockHeight)
+
+	resultChan := make(chan *blockResults, 100)
+	globalCtx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	// Fetching RPC data job
+	fetchingEg, jobCtx := errgroup.WithContext(globalCtx)
+	fetchingEg.SetLimit(6)
+
+	fetchingEg.Go(func() error {
+		clients := &clients{
+			celoEthclient: celoEthClient,
+			opEthclient:   opEthClient,
+			celoClient:    celoClient,
+			opClient:      opClient,
+		}
+
+		// Fetch some random blocks between 0 to lastCeloL1BlockHeight
+		for currentBlockNumber := uint64(0); currentBlockNumber < lastCeloL1BlockHeight; currentBlockNumber += blockInterval {
+			// exit loop if context is canceled
+			if isContextCanceled(jobCtx) {
+				t.Logf("Context canceled, exiting fetching loop at height %d", currentBlockNumber)
+				return nil
+			}
+
+			// decide block number to fetch between [currentBlockNumber, min(currentBlockNumber+blockInterval-1, lastCeloL1BlockHeight))
+			offset := uint64(0)
+			if enableRandomBlockTest {
+				randomUpperBound := blockInterval
+				if currentBlockNumber+randomUpperBound-1 >= lastCeloL1BlockHeight {
+					randomUpperBound = lastCeloL1BlockHeight - currentBlockNumber
+				}
+
+				// Int63n returns a non-negative random number
+				offset = uint64(rand.Int63n(int64(randomUpperBound)))
+			}
+
+			// Fetch data at the point of current range
+			asyncFetchRpcData(t, jobCtx, fetchingEg, currentBlockNumber+offset, clients, resultChan)
+		}
+
+		// Fetch data at the last block
+		asyncFetchRpcData(t, jobCtx, fetchingEg, lastCeloL1BlockHeight, clients, resultChan)
+
+		return nil
+	})
+
+	// Testing fetched data job
+	testingEg, jobCtx := errgroup.WithContext(globalCtx)
+	testingEg.Go(func() error {
+		for result := range resultChan {
+			if isContextCanceled(jobCtx) {
+				t.Logf("Context canceled, exiting testing loop at height %d", result.blockNumber)
+				return nil
+			}
+
+			result := result
+			testingEg.Go(func() error {
+				// Reset totalDifficulty of L1 block because totalDifficulty of L2 block is always 0x0
+				result.celoRawBlockByNumber["totalDifficulty"] = "0x0"
+				result.celoRawBlockByHash["totalDifficulty"] = "0x0"
```

Comment:
Hey @Kourin1996 where did you get this data from?

I see the `totalDifficulty` of L2 blocks being consistent with the L1 blocks.

```shell
export ETH_RPC_URL=https://alfajores-forno.celo-testnet.org
cast rpc eth_getBlockByNumber 0x2 true  | jq | grep totalDifficulty
  "totalDifficulty": "0x3",
```

Reply:
@piersy I see. local L2 seems to return wrong value. I run local L2 node by `celo-l2-node-docker-compose`/`main` and L1 node as well. Somehow local L2 node returns `0x0` for migrated blocks. (btw, This L2 node didn't have initial local data and download all data from peer)

I checked `https://alfajores-forno.celo-testnet.org` returned proper value. I think I can remove them

Reply:
Thanks @Kourin1996, I checked with a snap synced node from the baklava test and the total difficulty was also zero. Created an issue [here](https://github.com/celo-org/celo-blockchain-planning/issues/849) to investigate.